demo/admin/src/products/ManufacturerForm.tsx

```diff
@@ -227,54 +232,72 @@ export function ManufacturerForm({ id }: FormProps): React.ReactElement {
-                                    label={
-                                        <FormattedMessage id="manufacturer.address.alternativeAddress.street" defaultMessage="Alt-Address Street" />
-                                    }
-                                />
-                                <Field
-                                    fullWidth
-                                    name="address.alternativeAddress.streetNumber"
-                                    component={FinalFormInput}
-                                    type="number"
-                                    label={
-                                        <FormattedMessage
-                                            id="manufacturer.address.alternativeAddress.streetNumber"
-                                            defaultMessage="Alt-Address Street Number"
-                                        />
-                                    }
-                                />
-                                <Field
-                                    required
-                                    fullWidth
-                                    name="address.alternativeAddress.zip"
-                                    component={FinalFormInput}
-                                    type="number"
-                                    label={<FormattedMessage id="manufacturer.address.alternativeAddress.zip" defaultMessage="Alt-Address Zip" />}
-                                />
-                                <TextField
-                                    required
-                                    fullWidth
-                                    name="address.alternativeAddress.country"
-                                    label={
-                                        <FormattedMessage id="manufacturer.address.alternativeAddress.country" defaultMessage="Alt-Address Country" />
-                                    }
-                                />
-                            </FieldSet>
+                                {(props) => (
+                                    <FormControlLabel
+                                        label={
+                                            values?.useAlternativeAddress ? (
+                                                <FormattedMessage id="manufacturer.address.useAlternativeAddress.yes" defaultMessage="Yes" />
+                                            ) : (
+                                                <FormattedMessage id="manufacturer.address.useAlternativeAddress.no" defaultMessage="No" />
+                                            )
+                                        }
+                                        control={<FinalFormSwitch {...props} />}
+                                    />
+                                )}
+                            </Field>
+                            {values?.useAlternativeAddress && (
+                                <>
```

Comment:
This can also be achieved by using Field twice, which is imho cleaner (see https://codesandbox.io/s/lm4p3m92q from the official examples)